applications/exposure-checker/templates/deployment.yaml

```diff
@@ -0,0 +1,104 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Chart.Name }}
+  labels:
+    {{- include "exposure-checker.labels" . | nindent 4 }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      {{- include "exposure-checker.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        {{- include "exposure-checker.selectorLabels" . | nindent 8 }}
+    spec:
+      volumes:
+        - name: "butler-secrets-raw"
+          secret:
+            secretName: {{ include "exposure-checker.fullname" . }}
+        - name: "butler-secrets"
+          emptyDir:
+            sizeLimit: 50Mi
+        {{- with .Values.config.volumes }}
+        {{- . | toYaml | nindent 8 }}
+        {{- end }}
+      containers:
+      - name: {{ .Chart.Name }}
+        image: {{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion}}
+        imagePullPolicy: Always
+        env:
```

Comment:
Consider applying all the non-secret environment variables to one or more configMaps. This makes it easier to reason about them, view them all in one place, and improves reuse should the need arise.